### ISSUE TITLE
Dart SDK -> Engine Roller Failing (dart2js_info doesn't exist)

### ISSUE THREAD

**Issue by iskakaushik** [flutter/flutter#88281]
Most relevant part:

```
Because flutter_frontend_server depends on dart2js_info from path which doesn\'t exist (could not find package dart2js_info at "../../third_party/dart/third_party/pkg/dart2js_info"), version solving failed.\n'
```

Full Error:

```
The roller failed to create a CL with: failed pre-upload step: Error when running "gclient sync" in /data/src/flutter: Command exited with exit status 2: DEPOT_TOOLS_UPDATE=0 HOME=/home/skia LUCI_CONTEXT=/data/tmp/luci_context.329298250 PATH=/data/depot_tools:/cipd:/cipd/cipd_bin_packages:/cipd/cipd_bin_packages/bin:/cipd/cipd_bin_packages/cpython:/cipd/cipd_bin_packages/cpython/bin:/cipd/cipd_bin_packages/cpython3:/cipd/cipd_bin_packages/cpython3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin SKIP_GCE_AUTH_FOR_GIT=1 python /data/depot_tools/gclient.py sync --delete_unversioned_trees --force; Stdout+Stderr: ________ deleting 'src/third_party/dart/third_party/pkg/dart2js_info' in '/data' ________ running 'python3 src/tools/remove_stale_pyc_files.py src/tools' in '/data' ________ running 'python3 src/build/linux/sysroot_scripts/install-sysroot.py --arch=x64' in '/data' ________ running 'python3 src/build/linux/sysroot_scripts/install-sysroot.py --arch=arm64' in '/data' ________ running 'python3 src/flutter/tools/pub_get_offline.py' in '/data' '/data/src/third_party/dart/tools/sdks/dart-sdk/bin/pub get --offline' failed in 'src/flutter/flutter_frontend_server' with status 66: b'Resolving dependencies...\nBecause flutter_frontend_server depends on dart2js_info from path which doesn\'t exist (could not find package dart2js_info at "../../third_party/dart/third_party/pkg/dart2js_info"), version solving failed.\n' Error: Command 'python3 src/flutter/tools/pub_get_offline.py' returned non-zero exit status 1 in /data '/data/src/third_party/dart/tools/sdks/dart-sdk/bin/pub get --offline' failed in 'src/flutter/flutter_frontend_server' with status 66: b'Resolving dependencies...\nBecause flutter_frontend_server depends on dart2js_info from path which doesn\'t exist (could not find package dart2js_info at "../../third_party/dart/third_party/pkg/dart2js_info"), version solving failed.\n' . At deps_local.go:141 git_common.go:153 git_checkout.go:74 autoroller.go:489 autoroller.go:447 state_machine.go:284 state_machine.go:53 state_machine.go:168 state_machine.go:839 state_machine.go:852 state_machine.go:858 autoroller.go:781 autoroller.go:337 util.go:391 cleanup.go:57 asm_amd64.s:1371 The AutoRoll server is located here: https://autoroll.skia.org/r/dart-sdk-flutter-engine
```

**Comment by aam** [comment]
Created https://bugs.chromium.org/p/skia/issues/detail?id=12345 for this.

**Comment by zanderso** [comment]
`src/flutter/tools/pub_get_offline.py` is failing because `flutter_frontend_server` is looking for `dart2js_info` in `../../third_party/dart/third_party/pkg/dart2js_info`, but it isn't there anymore.

**Comment by aam** [comment]
> src/flutter/tools/pub_get_offline.py is failing because the flutter_frontend_server pubspec is looking for dart2js_info in ../../third_party/dart/third_party/pkg/dart2js_info, but it isn't there anymore.

`gclient sync` when run on the roller deletes dart2js_info folder:
```
Stdout+Stderr: ________ deleting 'src/third_party/dart/third_party/pkg/dart2js_info' in '/data'
```

This doesn't happen when run locally. 
Hopefully Ravi can take a look at https://bugs.chromium.org/p/skia/issues/detail?id=12345

**Comment by aam** [comment]
cc @rmistry

**Comment by alexmarkov** [comment]
https://dart-review.googlesource.com/c/sdk/+/210060 removed dependency on `dart2js_info` from Dart SDK DEPS.
`gclient sync -D` removes this directory locally for me as well - it is no longer needed by DEPS.

**Comment by zanderso** [comment]
The roll may have to be manual, and at the same time remove the reference to `dart2js_info` from the pubspec.

**Comment by aam** [comment]
Thanks for checking @alexmarkov , you are right, I was looking at older dart sdk revision.
cc @joshualitt

**Comment by aam** [comment]
https://github.com/flutter/engine/pull/28107 with manual roll

**Comment by a-siva** [comment]
Manual roll with fix from @aam has landed, closing issue.
